pkg/utils/nodeutils.go

```diff
@@ -121,6 +121,14 @@ func McfgPoolLabelMatches(nodeSelector map[string]string, pool *mcfgv1.MachineCo
 		return false
 	}
 	// TODO(jaosorior): Make this work with MatchExpression
+	if pool.Spec.NodeSelector == nil {
+		return false
+	}
+
+	if pool.Spec.NodeSelector.MatchLabels == nil {
+		return false
+	}
+
 	return reflect.DeepEqual(nodeSelector, pool.Spec.NodeSelector.MatchLabels)
 }
```

Comment:
It would be ideal to come up with a unit test for this to ensure we don't regress